sql/src/main/java/edu/stanford/futuredata/macrobase/sql/MacroBaseSQLRepl.java

```diff
@@ -0,0 +1,210 @@
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.impl.Arguments;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MacroBaseSQLRepl {
+
+    private static final String ASCII_ART_FILE = "ascii_art.txt";
+    private static final Logger log = LoggerFactory.getLogger(MacroBaseSQLRepl.class);
+
+    private final ConsoleReader reader;
+    private final SqlParser parser;
+    private final QueryEngine queryEngine;
+    private final boolean paging;
+
+    /**
+     * Main entry point to the SQL CLI interface in MacroBase
+     *
+     * @throws IOException if unable to instantiate ConsoleReader
+     */
+    private MacroBaseSQLRepl(final boolean paging) throws IOException {
+        // Initialize console reader and writer
+        this.paging = paging;
+        reader = new ConsoleReader();
+        final CandidateListCompletionHandler handler = new CandidateListCompletionHandler();
+        handler.setStripAnsi(true);
+        reader.setCompletionHandler(handler);
+        reader.addCompleter(new FileNameCompleter());
+
+        parser = new SqlParser();
+        queryEngine = new QueryEngine();
+    }
+
+    /**
+     * Executes one or more SQL queries.
+     *
+     * @param queries A single String which contains the queries to execute. Each query in the
+     * String should be delimited by ';' and optional whitespace.
+     * @param fromFile If True, queries have been read from File
+     */
+    private void executeQueries(final String queries, final boolean fromFile) {
+        StatementSplitter splitter = new StatementSplitter(queries);
+        for (StatementSplitter.Statement s : splitter.getCompleteStatements()) {
+            final String statementStr = s.statement();
+            if (fromFile) {
+                System.out.println(statementStr + ";");
+                System.out.println();
+                System.out.flush();
+            }
+            // Remove extra whitespace and add delimiter before updating console history
+            reader.getHistory().add(statementStr.replaceAll("\\s+", " ") + ";");
+            try {
+                Statement stmt = parser.createStatement(statementStr);
+                log.debug(stmt.toString());
+                final DataFrame result;
+                if (stmt instanceof ImportCsv) {
+                    final ImportCsv importStatement = (ImportCsv) stmt;
+                    result = queryEngine.importTableFromCsv(importStatement);
+                } else {
+                    final QueryBody q = ((Query) stmt).getQueryBody();
+                    result = queryEngine.executeQuery(q);
+                }
+                if (paging) {
+                    try {
+                        final PrintStream ps = new PrintStream(
+                            new FileOutputStream("/tmp/mb-sql.output"));
```

Comment:
This breaks on Windows and probably has permissions issues in Linux at least sometimes.   There's got to be a better way.